go-controller/pkg/ovn/external_ids_syncer/port_group/port_group_sync.go

```diff
@@ -0,0 +1,230 @@
+package port_group
+
+import (
+	"fmt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"strings"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// port groups suffixes
+	// ingressDefaultDenySuffix is the suffix used when creating the ingress port group for a namespace
+	ingressDefaultDenySuffix = "ingressDefaultDeny"
+	// egressDefaultDenySuffix is the suffix used when creating the ingress port group for a namespace
+	egressDefaultDenySuffix      = "egressDefaultDeny"
+	defaultNetworkControllerName = "default-network-controller"
+)
+
+type updatePortGroupInfo struct {
+	acls  []*nbdb.ACL
+	oldPG *nbdb.PortGroup
+	newPG *nbdb.PortGroup
+}
+
+type PortGroupSyncer struct {
+	nbClient libovsdbclient.Client
+	// txnBatchSize is used to control how many port groups will be updated with 1 db transaction.
+	// Set to 0 to disable batching
+	txnBatchSize int
+}
+
+func getControllerName(networkExternalID string) string {
```

Comment:
what is a networkExternalID? is that a secondary networks specific thing?

Reply:
yeah that is pre-dbIDs way to differentiate port groups managed by different controllers. It will only exist in syncer now to translate the old way to the new way